0MI0800304/28.2.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+struct OrderedPair{
+    int first{ 0 };
+    int second{ 0 };
+    void readOrderedPair() {
+        std::cin >> first;
+        std::cin >> second;
+    }
+    OrderedPair compare(OrderedPair& x2) const {
```

Comment:
It's rather atypical for a `compare` function to return an object. What's considered a standard practice is to return `a negative int` when `left is smaller than right`, `zero int` when `left is equal to right` and `a positive int` when `left is greater than right`